commit faa91915213d86b792d7fca491d199aea2243062
Author: Antenore Gatta <antenore@simbiosi.org>
Date:   Tue Dec 10 14:18:07 2019 +0100

    Adding cmake option to cuse latest FreeRDP symbols when compiling. Fixes #2024

diff --git a/plugins/rdp/CMakeLists.txt b/plugins/rdp/CMakeLists.txt
index 90c8966b7..866a015fd 100644
--- a/plugins/rdp/CMakeLists.txt
+++ b/plugins/rdp/CMakeLists.txt
@@ -60,6 +60,12 @@ add_definitions(-DFREERDP_REQUIRED_MAJOR=${FREERDP_REQUIRED_MAJOR})
 add_definitions(-DFREERDP_REQUIRED_MINOR=${FREERDP_REQUIRED_MINOR})
 add_definitions(-DFREERDP_REQUIRED_REVISION=${FREERDP_REQUIRED_REVISION})
 
+option(WITH_FREERDP_MASTER "Compile the RDP plugin using symbols from FreeRDP master branch." OFF)
+if(WITH_FREERDP_MASTER)
+	message(STATUS "Compiling the RDP plugin using symbols from FreeRDP master branch.")
+	add_definitions(-DWITH_FREERDP_MASTER=ON)
+endif()
+
 add_library(remmina-plugin-rdp MODULE ${REMMINA_PLUGIN_RDP_SRCS})
 set_target_properties(remmina-plugin-rdp PROPERTIES PREFIX "")
 set_target_properties(remmina-plugin-rdp PROPERTIES NO_SONAME 1)
diff --git a/plugins/rdp/rdp_plugin.c b/plugins/rdp/rdp_plugin.c
index 836823afd..ca57f6c30 100644
--- a/plugins/rdp/rdp_plugin.c
+++ b/plugins/rdp/rdp_plugin.c
@@ -1172,8 +1172,15 @@ static gboolean remmina_rdp_main(RemminaProtocolWidget *gp)
 		freerdp_set_gateway_usage_method(rfi->settings,
 						 remmina_plugin_service->file_get_int(remminafile, "gateway_usage", FALSE) ? TSC_PROXY_MODE_DETECT : TSC_PROXY_MODE_DIRECT);
 
+#ifdef WITH_FREERDP_MASTER
+#pragma message  "Using FreeRDP master branch!"
+	/* TODO: As soon as FreeRDP 2.0.0-rc5 will be available, implement an ifdef with that version */
 	freerdp_settings_set_string(rfi->settings, (size_t)FreeRDP_GatewayAccessToken,
 				    remmina_plugin_service->file_get_string(remminafile, "gatewayaccesstoken"));
+#else
+	rfi->settings->GatewayAccessToken = g_strdup(
+			remmina_plugin_service->file_get_string(remminafile, "gatewayaccesstoken"));
+#endif
 
 	rfi->settings->AuthenticationLevel = remmina_plugin_service->file_get_int(
 		remminafile, "authentication level", rfi->settings->AuthenticationLevel);
